crates/bevy_render/src/camera/visible_entities.rs

```diff
@@ -24,22 +24,200 @@ impl VisibleEntities {
     }
 }
 
+/// A mask describes which rendering group an entity belongs to.
+///
+/// Cameras with this component will only render entities with a matching
+/// mask.
+///
+/// There are 32 groups numbered `0` - `31`. A mask may belong to one or more
+/// groups, or no group at all.
+///
+/// An entity with a mask belonging to no groups is invisible.
+///
+/// The [`Default`] instance of `RenderingMask` returns a mask belonging to
+/// group `0`, the first group.
+#[derive(Copy, Clone, Reflect, PartialEq, Eq, PartialOrd, Ord)]
+#[reflect(Component)]
+pub struct RenderingMask(u32);
+
+impl std::fmt::Debug for RenderingMask {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("RenderingMask")
+            .field(&self.get_groups())
+            .finish()
+    }
+}
+
+impl std::iter::FromIterator<u8> for RenderingMask {
+    fn from_iter<T: IntoIterator<Item = u8>>(i: T) -> Self {
+        i.into_iter()
+            .fold(RenderingMask(0), |mask, g| mask.with_group(g))
+    }
+}
+
+/// Defaults to a mask belonging to group `0`, the first group.
+impl Default for RenderingMask {
+    fn default() -> Self {
+        RenderingMask(1)
+    }
+}
+
+impl RenderingMask {
```

Comment:
Imo "mask" is an implementation detail here. I think my preferred name is `RenderLayers` (and we could rename instances of "group" to "layer")

Reply:
In fact, this would let us simplify a bunch of method names:

* RenderMask::with_group(1) -> RenderLayers::with(1)
* RenderMask::all_groups() -> RenderLayers::all()

Reply:
Done.